src/lib/swapper/swappers/ThorchainSwapper/utils/addL1ToLongtailPartsToMemo.ts

```diff
@@ -0,0 +1,87 @@
+import { type AssetId, type ChainId } from '@shapeshiftoss/caip'
+import { bn } from '@shapeshiftoss/chain-adapters'
+import assert from 'assert'
+import BigNumber from 'bignumber.js'
+import type { Address } from 'viem'
+import { assertAndProcessMemo } from 'lib/utils/thorchain/memo'
+import { subtractBasisPointAmount } from 'state/slices/tradeQuoteSlice/utils'
+
+import { getMaxBytesLengthByChainId } from '../constants'
+import { MEMO_PART_DELIMITER } from './constants'
+import { getUniqueAddressSubstring } from './getUniqueAddressSubstring'
+import { shortenedNativeAssetNameByNativeAssetName } from './longTailHelpers'
+import { makeMemoWithShortenedFinalAssetAmount } from './makeMemoWithShortenedFinalAssetAmount'
+
+export const addL1ToLongtailPartsToMemo = ({
+  sellChainId,
+  quotedMemo,
+  aggregator,
+  finalAssetContractAssetId,
+  finalAssetAmountOut,
+  slippageBps,
+  longtailTokens,
+}: {
+  sellChainId: ChainId
+  slippageBps: BigNumber.Value
+  quotedMemo: string | undefined
```

Comment:
Is there any reason `quotedMemo` can be undefined here if we're calling this in a flow where we're mutating a quoted, defined memo?

Reply:
Weird, it can't be undefined, nice catch 🐺